src/nbl/video/IOpenGL_FunctionTable.h

```diff
@@ -13,7 +13,7 @@ namespace nbl {
 
 		// This class contains pointers to functions common in GL 4.6 and GLES 3.2
 		// And implements (at least a common part) extGl* methods which can be implemented with those pointers
-		class COpenGL_FunctionTableBase
+		class IOpenGL_FunctionTable
```

Comment:
because of Shitty OpenGL ES, and its lack of Direct State Access, you will need to add non-DSA workarounds, but please dont add them to the `exGLNamed` (DSA) functions defined here... unless you plan to take a template parameter `isES` and do an `if constexpr` so that DSA is still required for the desktop GL

Reply:
I think you don't understand purpose of this interface

Reply:
well it has the special DSA/extension handling fallback chains
https://github.com/Devsh-Graphics-Programming/Nabla/blob/6d00ebee58278d5cbc353061f3890645453f527d/src/nbl/video/COpenGLFunctionTable.h#L297

so the ES issue is relevant

Reply:
i don't understand what's the issue

desktop GL has dsa and fallback to non-dsa at the worst case
for ES there's non-dsa but I don't get it why wouldn't they be under extGl* interface. It's like 80% of existence sense of this interface

Reply:
well then take an `bool isES` template paramter, its just because the desktop backend shouldn't even be considering the non-DSA fallbacks and checking extension features

When it impacts every DSA function the cost of checking a branch quickly adds up

Reply:
no need for `isES`, just dont IOpenGL_FunctionTable::func in COpenGLFunctionTable functions will do

Reply:
!?

Reply:
in IOpenGL_FunctionTable theres common stuff, in case of dsa there are non-dsa workarounds

Reply:
ok but fail desktop device creation if DSA EXT, ARB or 4.5 isn't available